### PR TITLE
Improve E2E workflow

### PR DESCRIPTION
**Pull request by pbchekin**
* Increase job timeout, some jobs are 5+ hours long
* Create .env file in artifacts that contains commit ids for dependencies and other environment information. Specifically, variable `GITHUB_RUN_ID` allows to merge reports from all jobs in the matrix.
* Bigger matrix: 3x2x3 instead of 3x2 to make each individual job smaller.
* Set `fail-fast` to false. Failing jobs should not cancel other running jobs in the matrix.

### CODE REVIEW

File: .github/workflows/e2e.yml
Comment (whitneywhtsang) [inline .github/workflows/e2e.yml:1]: would guiding some of the steps by model save us some time? e.g., we don't need to install timm for huggingface or torchbench, we don't need to install torchtext for huggingface and timm_models, etc.

Comment (pbchekin) [reply .github/workflows/e2e.yml:1]: > would guiding some of the steps by model save us some time? e.g., we don't need to install timm for huggingface or torchbench, we don't need to install torchtext for huggingface and timm_models, etc.

Sure, could you please help with identifying such improvements? I have no idea what is needed and what is not for each suite.

Comment (pbchekin) [reply .github/workflows/e2e.yml:1]: Currently the overhead (installing everything using cache) is about 10 minutes per job, which is OK if e2e test itself takes 5+ hours. But if we can reduce the overhead for some cases, that would be perfect.

Comment (whitneywhtsang) [reply .github/workflows/e2e.yml:1]: Let's start with huggingface, as that's the easiest.
huggingface only needs `torch` from `Stonepia/pytorch`, `intel-extension-for-pytorch`, and `transformers`.

Comment (pbchekin) [reply .github/workflows/e2e.yml:1]: It still needs `weishi-deng/benchmark.git`, doesn't it?

Comment (whitneywhtsang) [reply .github/workflows/e2e.yml:1]: no, it is not needed for hugging face.

Comment (pbchekin) [reply .github/workflows/e2e.yml:1]: OK, done. I'll run the workflow today to check how it works.

File: .github/workflows/e2e.yml [hunk 1]
```diff
@@ -212,52 +231,57 @@ jobs:
...
+      - name: Install transformers for huggingface
+        if: ${{ matrix.suite == 'huggingface' }}
+        run: |
+          pip install transformers==$TRANSFORMERS_VERSION
+
       - name: Install PyTorch Benchmarks
+        if: ${{ matrix.suite == 'timm_models' || matrix.suite == 'torchbench' }}
         run: |
-          pip install pyyaml pandas scipy psutil pyre_extensions torchrec transformers==$TRANSFORMERS_VERSION
+          pip install pyyaml pandas scipy psutil pyre_extensions torchrec
```
Comment (whitneywhtsang) [inline .github/workflows/e2e.yml:284]: pandas is needed for huggingface too

Comment (pbchekin) [reply .github/workflows/e2e.yml]: Fixed

**Review by kurapov-peter — APPROVED** [review]

File: .github/workflows/e2e.yml [hunk 2]
```diff
@@ -212,52 +233,56 @@ jobs:
...
+        if: ${{ matrix.suite == 'timm_models' || matrix.suite == 'torchbench' }}
         run: |
           pip install audio/dist/*.whl
           python -c "import torchaudio; print(torchaudio.__version__)"
 
       - name: Install PyTorch Benchmarks
+        if: ${{ matrix.suite == 'timm_models' || matrix.suite == 'torchbench' }}
         run: |
-          pip install pyyaml pandas scipy psutil pyre_extensions torchrec transformers==$TRANSFORMERS_VERSION
+          pip install
```
Comment (pbchekin) [inline .github/workflows/e2e.yml:285]: Typo
```suggestion
          #pip install
```

File: .github/workflows/e2e.yml [hunk 3]
```diff
@@ -212,52 +233,56 @@ jobs:
...
+        if: ${{ matrix.suite == 'timm_models' || matrix.suite == 'torchbench' }}
         run: |
           pip install audio/dist/*.whl
           python -c "import torchaudio; print(torchaudio.__version__)"
 
       - name: Install PyTorch Benchmarks
+        if: ${{ matrix.suite == 'timm_models' || matrix.suite == 'torchbench' }}
         run: |
-          pip install pyyaml pandas scipy psutil pyre_extensions torchrec transformers==$TRANSFORMERS_VERSION
+          #pip install
```
Comment (whitneywhtsang) [inline .github/workflows/e2e.yml:285]: remove?

**Review by whitneywhtsang — APPROVED** [review]

**Pull request merged**